### PR TITLE
Implement Vision API for `Provider#complete`

### PR DESCRIPTION
**Pull request by antaz**
## Changes
- Added `Provider#transform_prompt` to handle URI-based image prompts
- Implemented image support via URI for `OpenAI#chat` and `Anthropic#chat`

## Examples
```ruby
image_url = "https://upload.wikimedia.org/wikipedia/commons/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg"

LLM.openai(OPENAI_KEY).chat(URI(image_url)).thread.last

#<LLM::Message:0x00007501ab6633a8
 @content=
  "This image depicts a serene path made of wooden planks winding through lush green grass and vegetation under a blue sky with soft white clouds. The scene evokes a sense of tranquility and connection to nature, inviting one to explore the landscape ahead. It captures the beauty of an open natural space, possibly a wetland or meadow, emphasizing the lushness of the environment.",
 @role="assistant">
```

### CODE REVIEW

File: lib/llm/providers/anthropic.rb [hunk 1]
```diff
@@ -44,5 +43,17 @@ def auth(req)
     def response_parser
       LLM::Anthropic::ResponseParser
     end
+
+    def transform_prompt(prompt)
+      return unless URI === prompt
```
Comment (0x1eef) [inline lib/llm/providers/anthropic.rb:48]: `nil` is not a valid prompt. Probably should be `return prompt unless URI === prompt`. Or:
```ruby
if URI === prompt
  # ...
else
  prompt
end
```

File: lib/llm/providers/openai.rb [hunk 2]
```diff
@@ -44,5 +43,11 @@ def auth(req)
     def response_parser
       LLM::OpenAI::ResponseParser
     end
+
+    def transform_prompt(prompt)
```
Comment (0x1eef) [inline lib/llm/providers/openai.rb:47]: This method can return `nil`. We should return `prompt` if no other condition matches.

**Review by 0x1eef — COMMENTED** [review]
Nice :+1: Left a few comments

File: lib/llm/providers/openai.rb (hunk 2 above)
Comment (antaz) [reply lib/llm/providers/openai.rb:47]: @0x1eef completely overlooked that! for some reason i thought just by returning prompt in the provider superclass we should be good, but that's obviously not the case.

File: lib/llm/providers/anthropic.rb [hunk 3]
```diff
@@ -43,5 +43,17 @@ def auth(req)
     def response_parser
       LLM::Anthropic::ResponseParser
     end
+
+    def transform_prompt(prompt)
```
Comment (0x1eef) [inline lib/llm/providers/anthropic.rb:47]: We should add references to Provider docs, Yard will follow them when generating docs. Eg:
```
##
# @param prompt (see LLM::Provider#transform_prompt)
# @return (see LLM::Provider#transform_prompt)

File: lib/llm/providers/openai.rb [hunk 4]
```diff
@@ -43,5 +43,11 @@ def auth(req)
...
```
Comment (0x1eef) [inline lib/llm/providers/openai.rb:47]: We should add references to Provider docs, Yard will follow them when generating docs. Eg:
```ruby
##
# @param prompt (see LLM::Provider#transform_prompt)
# @return (see LLM::Provider#transform_prompt)
```

File: lib/llm/provider.rb [hunk 5]
```diff
@@ -85,5 +86,15 @@ def preflight(req, body)
       auth(req)
       req
     end
+
+    ##
+    # Transforms the prompt before sending it
+    # @param [String] prompt
```
Comment (0x1eef) [inline lib/llm/provider.rb:92]: `String` doesn't really cover it. `String,URI,Object` would cover all base.

**Review by 0x1eef — APPROVED** [review]
Nice :+1: Look forward to testing this

**Pull request merged**